gromacs/gromacs-mdmodules/applied_forces/colvars/colvarsforceprovider.h

```diff
@@ -0,0 +1,210 @@
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ *
+ * GROMACS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with GROMACS; if not, see
+ * https://www.gnu.org/licenses, or write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA.
+ *
+ * If you want to redistribute modifications to GROMACS, please
+ * consider that scientific software is very special. Version
+ * control is crucial - bugs must be traceable. We will be happy to
+ * consider code for inclusion in the official distribution, but
+ * derived work must not be called official GROMACS. Details are found
+ * in the README & COPYING files - if they are missing, get the
+ * official version at https://www.gromacs.org.
+ *
+ * To help us fund GROMACS development, we humbly ask that you cite
+ * the research papers on the package. Check out https://www.gromacs.org.
+ */
+/*! \internal \file
+ * \brief
+ * Declares the force provider for colvars
+ *
+ * \author Hubert Santuz <hubert.santuz@gmail.com>
+ * \ingroup module_applied_forces
+ */
+
+#ifndef GMX_APPLIED_FORCES_COLVARSFORCEPROVIDER_H
+#define GMX_APPLIED_FORCES_COLVARSFORCEPROVIDER_H
+
+
+#include "gromacs/domdec/localatomset.h"
+#include "gromacs/mdrunutility/mdmodulesnotifiers.h"
+#include "gromacs/mdtypes/iforceprovider.h"
+
+#include "colvarproxygromacs.h"
+
+
+namespace gmx
+{
+
+
+/*! \internal
+ * \brief Parameters defining the internal colvars force provider state.
+ */
+struct ColvarsForceProviderState
+{
+
+    /*! \brief Indicate if a colvars state was read.
+     */
+    bool stateRead_ = false;
+
+    /*! \brief The number of colvars atoms.
+     */
+    std::int64_t nColvarsAtoms_ = 0;
+
+    /*! \brief String naming variable holding the number of colvars atoms.
+     * \note Changing this name will break backwards compability for checkpoint file writing.
+     */
+    static const std::string nColvarsAtomsName_;
+
+    //! Last known whole positions of the colvars atoms
+    //! \todo Change the type to a standard one to avoid memory leak.
+    rvec* xOldWhole_ = nullptr;
+
+    /*! \brief String naming variable holding the last known whole positions of the colvars atoms
+     * \note Changing this name will break backwards compability for checkpoint file writing.
+     */
+    static const std::string xOldWholeName_;
+
+    /*! \brief Content of the unformatted Colvars state file.
+     */
+    std::vector<unsigned char> colvarStateFile_;
+
+    /*! \brief String naming variable holding the content of the formatted Colvars state file.
```

Comment:
"formatted" here and below whereas "unformatted" is written just above

Reply:
Done, thank you!